### DISCUSSION TITLE
Suggestion: Force object detection to run at intervals

### DISCUSSION THREAD

**Discussion by nicoleise** [blakeblackshear/frigate#12933] [Ideas]
This is a feature request, unless Frigate already does this - haven't been able to find anything on it. 

Is it possible to configure Frigate to periodically run object detection on the camera stream, even if no motion is detected?

Either (preferably) by declaring it in the config, or by some external trigger?

For example, I use Frigate extensively to control lighting, a virtual doorbell (doorbell is sounded by presence in an area in front of a mock doorbell that doesn't actually work), etc. I also noticed an issue posted here, where someone uses Frigate to notify them of a street parking spot outside their house becoming free, and where Frigate would say the spot is free when it wasn't. 

In all these occupancy cases, Frigate can falsely report no occupancy even though detectable objects are in view, if there's a lack of motion over a long enough timeframe. 

For the doorbell scenario, this is inconsequential. 

However, for the lighting controls (I use it both indoor and outside, for example in my office), it matters a lot because the light will randomly turn off. And for the street parking issue I found, a lack of motion would also lead Frigate to a report that no car is present, taking up the spot, even though it is. 

This can be mended somewhat by using `for` in automations, but it to avoid false negatives it would need to be set to pointlessly large values. For example, my lighting in the office is set to turn off after 15 minutes of no motion detected, but it still turns off a few times a day, when I'm in plain view, typing away on some email. The same for the parking spot - if the street is quiet enough, you'd need a delay so long someone else might park in the spot before you get notified.

It would seem a simple and better solution to allow the object detection to be triggered at a specified interval in addition to be triggered by motion, especially with v0.14 seemingly being much better at handling overlapping events. Thus it shouldn't bring about any unwarranted "noise". 

Something like `periodic_detection: 5` to specify a mandatory object detection every 5 minutes regardless of motion is detected. 

Frigate should potentially cascade/waterfall this so that the event interval isn't triggered simultaneously on every camera with the option specified, but rather one by one. 

Does it seem like something that makes sense? And does Frigate potentially already support this somehow?

**Comment by NickM-27** [comment]
Frigate not only detects but also continuously tracks objects. Including checking if a stationary object is still there. If the object stops being detected incorrectly this is because the model isn't detecting the object confidently or at all, so running detection at a random time wouldn't help detect it. 

In general lowering required scores or using a model better suited for the circumstances is the best solution
